### PR TITLE
Csum add option

### PR DESCRIPTION
**Pull request by jjgomezcadenas**
Add option to calibrated_pmt_sum

Current version of calibrated_pmt_sum assumes that all the PMTs in the
energy plane are active. This is not necessarily true. This PR
modifies the function to allow passing  a list of active_pmt_index. Such list
specifies the indexes of active PMTs (PMT indexes run from 0 to 11). If the list is specified (e.g, not empty), only the active PMTs are considered for the sum. 
if the list is empty all 12 PMTs are considered for the sum, as before.

Tests have been expanded/updated to demonstrate the new functionality

### CODE REVIEW

**Comment by jjgomezcadenas** [comment]
This pull request was place accidentally on top of a branch that contains work from a different stream. The only relevant commits are the last two, thus I am closing it and will resubmit a cleaner PR

**Comment by jacg** [comment]
>  I am closing it and will resubmit a cleaner PR
 
Why !??!?!??!!!!!!!1111??

Please, pretty please with sugar on top: **stop opening and closing pull requests cada vez que te levantas a mear**. That's good practice for commits in your local working tree (or even your local fork on GitHub), but awful practice for PRs.

Instead of closing this one and submitting a clearer one, you could (and should) have simply pushed the cleaned up version onto this PR.

PRs are an invitation to discuss some code. Your habit of pulling the plug on these conversations and then restarting them somewhere else, really doesn't help the smooth exchange of information.

**Comment by jjgomezcadenas** [comment]
Sorry master!!!!

I thought no one had looked yet into it, so I could get away without a reprimand!

Not an habit, really, only a compulsion. Will not happen again (I hope)!

**Comment by jacg** [comment]
If no one had looked at it yet, you could have got away without attracting attention to its messy state, by simply pushing the change. Even if many people had looked at it, a push to this PR would have been the way to go, though, as a matter of courtesy, you might point out that you are rebasing. But even that is unnecessary, I think: we have incorporated these rebases into our standard workflow and it should be clear to everyone what is going on, without having to shout about it every single time.

Perhaps if you can explain to me what the appeal is of closing PRs and opening them elsewhere, I might be able to help in some way.

**Comment by jjgomezcadenas** [comment]
I think it has to do with incomplete understanding of the way github works. Plus panicking at being caught in a mess. Sorry. J

**Comment by jacg** [comment]
> I think it has to do with incomplete understanding of the way github works.

May I recommend that you start reading and posting on GitHub *on GitHub*, rather than in your email? I think such a practice would help you understand GitHub more, not to mention helping you avoid a number of problems I have observed which arise as a consequence of you Githubbing in your email. To take but one example: the poor guy we spammed last week? I haven't checked the exact details, but that was almost certainly a consequence of someone mistyping my github id in an email client. Making this mistake in GitHub is much more difficult, because GitHub conveniently prompts and completes names which are relevant to the project or conversation, so you really have to go out of your way to get a name wrong, and spam a poor innocent soul with our crap.

In summary, I recommend reading and writing GitHub messages on GitHub, not in your email.

**Comment by jjgomezcadenas** [comment]
The problem with GitHub is that things are not centralised. For example this thread comes in a closed PR (yes, sorry!). E-mail is centralised thus, one tends to answer on the fly to the various issues, without the need to navigate through the corresponding pages in GitHub. Although I agree with you, of course, that GitHub is the right place for the threads, the e-mail has the particular advantage of centralisation

**Comment by jacg** [comment]
Hmm, I disagree. On GitHub, the things which belong together, are grouped together (except when terrorists disperse them by closing PRs and re-opening them, etc.) Email is centralized in the sense that you don't have to move away from the window which happens to be focused on your screen, but crucial *information* is missing from that window. On numerous occasions you have made comments or asked questions which would never have arisen had you been using GitHub. On GitHub relevant information is right there in front of you, or but a small scroll away. It encourages you to review the conversation so far rather than ignoring it, in a way that your email window does not.

In theory, you could use your email client to access that information: if you have not deleted the email, and your email client's thread-handling abilities are top-notch, then you could access this information easily. In practice that seems not to happen.

Also, any GitHub email contains the link to the conversation, so it's just a question of following that link, and there you have all the information in front of you. I have watched you struggle with the numerous browser tabs that this opens for you, but I think that it wouldn't take too much practice to get into the habit of closing the tab once you've finished with it.

I've just started the practice of dedicating a virtual screen to each issue or PR. Each with its editor showing me the relevant files, and its browser showing the relevant GitHub page. Then I have an IC screen, where I run the tests. Of course, my window manager (not to mention my editor) is optimized for this sort of thing, because I went out of my way to optimize it for useful workflows ... because my time is too precious to waste on the defaults that come out of the box on current computer systems.

It has been observed that people who go out of their way to find better-than-default approaches to tasks are more productive (statistically speaking). I suspect that you use email to GitHub, not because you evaluated a number of alternative possibilites and decided that this is the best one, but rather because it's the default: notification from GitHub arrives in your email, so you respond to it in your email. I further submit to you that if you make a little effort to move away from this default, you (and those interacting with you) will be more productive on GitHub.

Which is a perfect introduction leading up to a rant on how anti-productive the (default-following) practice of top-posting is ... but I'll resist that temptation.

**Comment by jjgomezcadenas** [comment]
Understood master. GitHub it is.
